src/main/scala/com/fulcrumgenomics/vcf/FixVcfPhaseSet.scala

```diff
@@ -0,0 +1,250 @@
+    case class UpdatedPhaseAndValue       (genotype: Genotype) extends Result { val description: String = "were set to phased and had phase set value updated" }
+    case class UpdatedPhaseOnly           (genotype: Genotype) extends Result { val description: String = "were set to phased" }
+    case class PhasedMissingPhaseSetValue (genotype: Genotype) extends Result { val description: String = "were phased but missing a phase set value" }
+    case class NotPhasedWithPhaseSetValue (genotype: Genotype) extends Result { val description: String = "were not phased but had a phase set value" }
+    case class NotPhasedNoPhaseSetValue   (genotype: Genotype) extends Result { val description: String = "were not phased and no phase set value" }
+    override def values: immutable.IndexedSeq[Result] = findValues
+  }
+}
+
+class VcfPhaseSetUpdater(header: VcfHeader, keepOriginal: Boolean, phaseGenotypesWithPhaseSet: Boolean) extends LazyLogging {
+  import VcfPhaseSetUpdater._
+  import VcfPhaseSetUpdater.Result._
+
+  private val phaseSetToPositionBySample: Map[String, mutable.HashMap[String, Int]] = {
+    header.samples.map { sample => (sample, scala.collection.mutable.HashMap[String, Int]()) }.toMap
+  }
+  private var lastChrom: String = ""
+  private var variants: Long    = 0
+  private var genotypes: Long   = 0
+  val descriptionCounter: SimpleCounter[String] = new SimpleCounter()
+
+  def update(variant: Variant): Variant = {
+    // Phase sets cannot span contigs!  So empty our mappings if we get to a new one
+    if (variant.chrom != lastChrom) {
+      this.phaseSetToPositionBySample.values.foreach(_.clear())
+      this.lastChrom = variant.chrom
+    }
+
+    variants += 1
+    val genotypes: Map[String, Genotype] = variant
+      .genotypes
+      .map { case (sampleName: String, genotype: Genotype) =>
+        // update the genotype
+        val result = updateGenotype(variant=variant, genotype=genotype)
+        // log the results
+        result match {
+          case NotPhasedWithPhaseSetValue(_)   =>
+            logger.warning(
+              "Genotype had a phase set but was unphased:" +
+                f"${variant.chrom}:${variant.pos}:${variant.id.getOrElse(".")} PS=${genotype("PS")}" +
+                "; consider r-running using `-x/--phase-genotypes-with-phase-set`"
+            )
+          case PhasedMissingPhaseSetValue(_) =>
+            logger.warning(
+              f"Genotype had no phase set but was phased: ${variant.chrom}:${variant.pos}:${variant.id.getOrElse(".")}"
+            )
+          case _                  => () // do nothing
+        }
+        this.descriptionCounter.count(result.description)
+        // return the (potentially) new/updated genotype
+        (sampleName, result.genotype)
+      }
+    variant.copy(genotypes=genotypes)
+  }
+
+  def updateGenotype(variant: Variant, genotype: Genotype): Result = {
+    this.genotypes += 1
+
+    (genotype.phased, genotype.get[Any]("PS")) match {
```

Comment:
I think that since you updated the header you should be able to use `get[String]` reliably here.  If that doesn't can you let me know please because that should get fixed.